client/src/ProtectedRoutes/ProtectedRoutes.jsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  Navigate,
+} from 'react-router-dom';
+import { useAuth } from '../Context/AuthContext';
+
+export function ProtectedAdmin({ children }) {
+  const { user } = useAuth();
+  if (!user.user) {
```

Comment:
you should check the role for the admin

Reply:
in the useAuth hook change it and store just the user in the user state
you are calling `user.user`
keep it simple and make it like this 
```suggestion
  if (!user) {
```